File: cpp/program19.cpp
// program70.c

#include <iostream>
using namespace std;

class Digit
{
public:
    int iNo;

    Digit(int X)
    {
        iNo = X;
    }

    int MaximumDigit()
    {
        int iDigit = 0;
        int iMax = 0;
        int iTemp = iNo;

        if(iNo < 0)
        {
            iNo = -iNo;
        }

        while (iTemp != 0)
        {
            iDigit = iTemp % 10;
            if(iDigit > iMax)
            {
                iMax = iDigit;
            }
            if(iMax == 9)
            {
                break;
            }
            iTemp = iTemp / 10;
        }
        return iMax;
    }
};

int main()
{
    int iValue = 0;
    int iRet = 0;

    cout<<"Enter number : "<<"\n";
    cin>>iValue;

    Digit dobj(iValue);

    iRet = dobj.MaximumDigit();
    cout<<"Maximum digit is : "<<iRet<<"\n";

    return 0;
}

File: java/program19.java
import java.util.*;

class Number
{
    public int iNo;

    public Number(int i)
    {
        iNo = i;
    }

    public int MaximumDigit()
    {
        int iDigit = 0;
        int iMax = 0;
        int temp = iNo;

        while(temp != 0)
        {
            iDigit = temp % 10;
            if(iMax < iDigit)
            {
                iMax = iDigit;
            }
            temp = temp / 10;
        }
        return iMax;
    }
}

class program19
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number ");
        int iValue = sobj.nextInt();

        Number obj = new Number(iValue);
        System.out.println("Maximum digit is : "+obj.MaximumDigit());
    }
}